### PR TITLE
Implement basic global search

### PR DESCRIPTION
**Pull request by lukicenturi**
Apr-28-2022 00-00-37

### CODE REVIEW

**Comment by lukicenturi** [comment]
Basic part of #1219

**Review by LefterisJP — COMMENTED** [review]
I get frontend exceptions when trying to run this branch:

```
vue.runtime.esm.js:619 [Vue warn]: Error in mounted hook (Promise/async): "TypeError: Cannot read properties of undefined (reading 'startsWith')"

found in

---> <GlobalSearch> at src/components/GlobalSearch.vue
       <VAppBar>
         <VApp>
           <App> at src/App.vue
             <Root>
```

and


```
vue.runtime.esm.js:1897 TypeError: Cannot read properties of undefined (reading 'startsWith')
    at ElectronInterop._callee6$ (electron-interop.ts:101:41)
    at tryCatch (runtime.js:63:15)
    at Generator.invoke [as _invoke] (runtime.js:294:1)
    at Generator.eval [as next] (runtime.js:119:1)
    at asyncGeneratorStep (electron-interop.ts:11:103)
    at _next (electron-interop.ts:13:194)
```

Other than that seems pretty cool as a basic fucntionality for this and we can eventually close the issue and if more is needed different issues should be made.

Kelsos should also check the code functionality ofc.

**Comment by lukicenturi** [comment]
> I get frontend exceptions when trying to run this branch:
> 
> ```
> vue.runtime.esm.js:619 [Vue warn]: Error in mounted hook (Promise/async): "TypeError: Cannot read properties of undefined (reading 'startsWith')"
> 
> found in
> 
> ---> <GlobalSearch> at src/components/GlobalSearch.vue
>        <VAppBar>
>          <VApp>
>            <App> at src/App.vue
>              <Root>
> ```
> 
> and
> 
> ```
> vue.runtime.esm.js:1897 TypeError: Cannot read properties of undefined (reading 'startsWith')
>     at ElectronInterop._callee6$ (electron-interop.ts:101:41)
>     at tryCatch (runtime.js:63:15)
>     at Generator.invoke [as _invoke] (runtime.js:294:1)
>     at Generator.eval [as next] (runtime.js:119:1)
>     at asyncGeneratorStep (electron-interop.ts:11:103)
>     at _next (electron-interop.ts:13:194)
> ```
> 
> Other than that seems pretty cool as a basic fucntionality for this and we can eventually close the issue and if more is needed different issues should be made.
> 
> Kelsos should also check the code functionality ofc.

Error is fixed

File: frontend/app/src/components/GlobalSearch.vue [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+        <v-icon>mdi-magnify</v-icon>
+      </menu-tooltip-button>
+    </template>
+    <div :class="$style.wrapper">
+      <v-autocomplete
+        ref="input"
+        v-model="selected"
+        no-filter
+        filled
+        no-data-text="No actions found"
```
Comment (kelsos) [inline frontend/app/src/components/GlobalSearch.vue:29]: Can we extract this to the en.json

File: frontend/app/src/components/GlobalSearch.vue [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+        :search-input.sync="search"
+        :background-color="dark ? 'black' : 'white'"
+        hide-details
+        :items="filteredItems"
+        auto-select-first
+        prepend-inner-icon="mdi-magnify"
+        append-icon=""
+        placeholder="Search anything..."
```
Comment (kelsos) [inline frontend/app/src/components/GlobalSearch.vue:37]: Same here

**Review by kelsos — COMMENTED** [review]
@lukicenturi looks great, I have only two minor comments

Another thing is that for me Ctrl + / doesn't seem to work to open the menu. It could be that the OS or the desktop environment overrides it, @LefterisJP did you check the shortcut does it work for you?

**Comment by LefterisJP** [comment]
> @lukicenturi looks great, I have only two minor comments
> 
> Another thing is that for me Ctrl + / doesn't seem to work to open the menu. It could be that the OS or the desktop environment overrides it, @LefterisJP did you check the shortcut does it work for you?

No idea there was a shortcut even. What is this for? So if I press `ctrl + /` it should open up the global search? I can try it.

**Comment by lukicenturi** [comment]
Cmd + / for mac.
Ctrl + / for non mac.

I've tried on mac and it's worked

**Comment by LefterisJP** [comment]
> Cmd + / for mac. Ctrl + / for non mac.
> 
> I've tried on mac and it's worked

For me `Windows key + /` worked for Linux.

**Comment by kelsos** [comment]
> > Cmd + / for mac. Ctrl + / for non mac.
> > I've tried on mac and it's worked
> 
> For me `Windows key + /` worked for Linux.

Yes, this seems to work fine. @lukicenturi any idea why? (on my keyboard windows key maps to cmd on macOS so maybe this could be related)

**Comment by LefterisJP** [comment]
> > > Cmd + / for mac. Ctrl + / for non mac.
> > > I've tried on mac and it's worked
> > 
> > 
> > For me `Windows key + /` worked for Linux.
> 
> Yes, this seems to work fine. @lukicenturi any idea why? (on my keyboard windows key maps to cmd on macOS so maybe this could be related)

Windows is the super key by default. So perhaps it's not ctr but super for linux. I wonder what would work in Windows itself :)

**Comment by lukicenturi** [comment]
> > > Cmd + / for mac. Ctrl + / for non mac.
> > > I've tried on mac and it's worked
> > 
> > 
> > For me `Windows key + /` worked for Linux.
> 
> Yes, this seems to work fine. @lukicenturi any idea why? (on my keyboard windows key maps to cmd on macOS so maybe this could be related)

If You're using Mac, and using an external keyboard, then yeah it worked (the modifier key used is the one that you use for Copying (Cmd + C), Pasting (Cmd + V), or other common commands.

**Comment by LefterisJP** [comment]
> > > > Cmd + / for mac. Ctrl + / for non mac.
> > > > I've tried on mac and it's worked
> > > 
> > > 
> > > For me `Windows key + /` worked for Linux.
> > 
> > 
> > Yes, this seems to work fine. @lukicenturi any idea why? (on my keyboard windows key maps to cmd on macOS so maybe this could be related)
> 
> If You're using Mac, and using an external keyboard, then yeah it worked (the modifier key used is the one that you use for Copying (Cmd + C), Pasting (Cmd + V), or other common commands.
> 
> The same is happens for windows and linux, it must be the common modifier key. Maybe u map your Windows key to Ctrl ? @LefterisJP (e.g, for coyping you press WIn + C)

nope I don't

**Comment by lukicenturi** [comment]
> > > > > Cmd + / for mac. Ctrl + / for non mac.
> > > > > I've tried on mac and it's worked
> > > > 
> > > > 
> > > > For me `Windows key + /` worked for Linux.
> > > 
> > > 
> > > Yes, this seems to work fine. @lukicenturi any idea why? (on my keyboard windows key maps to cmd on macOS so maybe this could be related)
> > 
> > 
> > If You're using Mac, and using an external keyboard, then yeah it worked (the modifier key used is the one that you use for Copying (Cmd + C), Pasting (Cmd + V), or other common commands.
> > The same is happens for windows and linux, it must be the common modifier key. Maybe u map your Windows key to Ctrl ? @LefterisJP (e.g, for coyping you press WIn + C)
> 
> nope I don't

Fixed. It should work with Ctrl + / in linux now

**Review by kelsos — APPROVED** [review]
lgtm @lukicenturi thank you :)

**Pull request merged**